R/analyse_strat.R

```diff
@@ -97,34 +106,38 @@ analyse_strat <- function(data = NULL, Ts, Tb, dates, H_ice = NULL, drho = 0.1,
     strat_doys <- alt_doys
   }
 
-  s_strat <- (rho_water(t=Tb) - rho_water(t=Ts)) >= drho & Ts > Tb # logical whether stratified at each time step
+  s_strat <- (rho_water(t = Tb) - rho_water(t = Ts)) >= drho & Ts > Tb # logical whether stratified
+  # at each time step
   # s_strat <- Ts - Tb  > dT # logical whether stratified at each time step
 
-  i_s_st <- diff(c(s_strat[1],s_strat))==1 # indices of stratification onset
-  i_s_en <- diff(c(s_strat[1],s_strat))==-1 # indices of stratification end
-  if(s_strat[1]) i_s_st <- c(NA, i_s_st) # if stratified at beginning of simulation, make first date NA
-  if(s_strat[length(s_strat)]) i_s_en <- c(i_s_en, NA) # if stratified at end of sim, set last strat date to NA
+  i_s_st <- diff(c(s_strat[1], s_strat)) == 1 # indices of stratification onset
+  i_s_en <- diff(c(s_strat[1], s_strat)) == -1 # indices of stratification end
+  if(s_strat[1]) i_s_st <- c(NA, i_s_st) # if stratified at beginning of simulation, make first
+  # date NA
+  if(s_strat[length(s_strat)]) i_s_en <- c(i_s_en, NA) # if stratified at end of sim, set last
+  # strat date to NA
   s_start <- dates[i_s_st] # summer strat start dates
   s_end   <- dates[i_s_en] # summer strat end dates
   # if(sum(s_strat)==0) s_start <- s_end <- dates[1] # if never stratifies, set to time=0
-  s_dur   <- as.double(difftime(s_end, s_start, units="days")) # duration of summer strat periods
+  s_dur   <- as.double(difftime(s_end, s_start, units = "days")) # duration of summer strat periods
 
-  a1 <- data.frame(year=strat_yrs[i_s_st],
-                   start=s_start, end=s_end, dur=s_dur,
+  a1 <- data.frame(year = strat_yrs[i_s_st],
+                   start = s_start, end = s_end, dur = s_dur,
                    startday = strat_doys[i_s_st],
                    endday = strat_doys[i_s_en])
   a1 <- subset(a1, year %in% yrs)
 
   s.max <- s.mean <- s.tot <- s.on <- s.off <-
     s.first <- s.last <- yr <- NULL
-  for(mm in unique(a1$year[!is.na(a1$year)])) { # remove NAs which are generated when the lake is stratified at the satrt or end of the simulation
-    a2 <- subset(a1, year==mm)
+  for(mm in unique(a1$year[!is.na(a1$year)])) { # remove NAs which are generated when the lake is
+    # stratified at the satrt or end of the simulation
+    a2 <- subset(a1, year == mm)
     ind <- which.max(a2$dur)
-    if(nrow(a2)==1) if(is.na(a2$dur)) ind <- NA # fixes issue if stratified at end of data period
-    yr <- c(yr,mm)
-    s.max <- c(s.max,max(a2$dur))
-    s.mean <- c(s.mean,mean(a2$dur))
-    s.tot <- c(s.tot,sum(a2$dur))
+    if(nrow(a2) == 1) if(is.na(a2$dur)) ind <- NA # fixes issue if stratified at end of data period
```

Comment:
This was already in the old code, but I'm not a fan of this if-statement in an if-statement on one line. Feel free to keep if it works, but I'd split it over multiple lines

Reply:
This is all legacy code from Tom